extensions/blocks/dialogue/edit.js

```diff
@@ -0,0 +1,260 @@
+	const {
+		participant,
+		participantSlug,
+		timeStamp,
+		content,
+		placeholder,
+	} = attributes;
+	const [ isFocusedOnParticipantLabel, setIsFocusedOnParticipantLabel ] = useState( false );
+
+	// Block context integration.
+	const participantsFromContext = context[ 'jetpack/conversation-participants' ];
+	const showTimeStamp = context[ 'jetpack/conversation-showtimestamp' ];
+
+	// Participants list.
+	const participants = participantsFromContext?.length ? participantsFromContext : defaultParticipants;
+
+	// Participant object.
+	const isCustomParticipant = !! participant && ! participantSlug;
+	const currentParticipantSlug = ! participant && ! participantSlug ? defaultParticipantSlug : participantSlug;
+	const currentParticipant = getParticipantBySlug( participants, currentParticipantSlug );
+	const participantLabel = isCustomParticipant ? participant : currentParticipant?.participant;
+
+	// Conversation context. A bridge between dialogue and conversation blocks.
+	const transcritionBridge = useContext( ConversationContext );
+
+	const baseClassName = 'wp-block-jetpack-dialogue';
+
+	/**
+	 * Helper to check if the gven style is set, or not.
+	 * It handles local and global (conversation) level.
+	 *
+	 * @param {string} style - style to check.
+	 * @returns {boolean} True if the style is defined. Otherwise, False.
+	 */
+	function hasStyle( style ) {
+		if ( isCustomParticipant || ! participantsFromContext ) {
+			return attributes?.[ style ];
+		}
+
+		return currentParticipant?.[ style ];
+	}
+
+	/**
+	 * Helper to toggle the value of the given style
+	 * It handles local and global (conversation) level.
+	 *
+	 * @param {string} style - style to toggle.
+	 * @returns {void}
+	 */
+	function toggleParticipantStyle( style ) {
+		if ( isCustomParticipant || ! participantsFromContext ) {
+			return setAttributes( { [ style ]: ! attributes[ style ] } );
+		}
+
+		transcritionBridge.updateParticipants( {
+			participantSlug: currentParticipantSlug,
+			[ style ]: ! currentParticipant[ style ],
+		} );
+	}
+
+	/**
+	 * Helper to build the CSS classes for the participant label.
+	 * It handles local and global (conversation) level.
+	 *
+	 * @returns {string} Participant CSS class.
+	 */
+	function getParticipantLabelClass() {
+		if ( isCustomParticipant || ! participantsFromContext ) {
+			return classnames( `${ baseClassName }__participant`, {
+				[ 'has-bold-style' ]: attributes?.hasBoldStyle,
+				[ 'has-italic-style' ]: attributes?.hasItalicStyle,
+				[ 'has-uppercase-style' ]: attributes?.hasUppercaseStyle,
+			} );
+		}
+
+		return classnames( `${ baseClassName }__participant`, {
+			[ 'has-bold-style' ]: currentParticipant?.hasBoldStyle,
+			[ 'has-italic-style' ]: currentParticipant?.hasItalicStyle,
+			[ 'has-uppercase-style' ]: currentParticipant?.hasUppercaseStyle,
+		} );
+	}
+
+	return (
+		<div className={ className }>
+			<BlockControls>
+				{ currentParticipant && isFocusedOnParticipantLabel && (
+					<ToolbarGroup>
```

Comment:
I'm not convinced that each individual dialogue block needs the bold/italics/caps settings--is there a specific use case you imagine that someone would want to control these individually per speaker?

<img width="350" alt="image" src="https://user-images.githubusercontent.com/1699996/101437252-5e869580-38d5-11eb-8443-92419f265aa4.png">

Reply:
It applies format individually if the dialogue block has a custom participant label:

<img width="395" alt="Screen Shot 2020-12-08 at 5 08 50 AM" src="https://user-images.githubusercontent.com/77539/101456926-8a287080-3913-11eb-9c6f-c37a7f1c3102.png">

But when the participant is defined by the conversation, the styles are applied globally.

Reply:
Would the ability to add a participant on a dialogue block that's specific to that block, be potentially confusing? This could lead to participants being defined in both places, and then not being able to set the styles correctly at the conversation level.